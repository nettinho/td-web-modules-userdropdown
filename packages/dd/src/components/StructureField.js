import React from "react";
import PropTypes from "prop-types";
import { Header, Container, Segment } from "semantic-ui-react";
import { connect } from "react-redux";
import StructureFieldProperties from "./StructureFieldProperties";
import StructureFieldForm from "./StructureFieldForm";

export const StructureField = ({ name, structureFieldLoading }) =>
  structureFieldLoading ? null : (
    <React.Fragment>
      <Container as={Segment} text>
        <Header as="h2">
          <Header.Content>{name}</Header.Content>
        </Header>
        <StructureFieldForm />
        <StructureFieldProperties />
      </Container>
    </React.Fragment>
  );

StructureField.propTypes = {
  name: PropTypes.string,
  structureFieldLoading: PropTypes.bool
};

const mapStateToProps = ({ structureField, structureFieldLoading }) => ({
  structureFieldLoading,
  ...structureField
});

export default connect(mapStateToProps)(StructureField);
