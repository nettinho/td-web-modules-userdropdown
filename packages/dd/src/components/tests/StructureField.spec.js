import React from "react";
import { shallow } from "enzyme";
import { StructureField } from "../StructureField";

describe("<StructureField />", () => {
  it("matches the latest snapshot", () => {
    const structureField = {
      id: 1,
      name: "f1",
      description: "dd",
      type: "tp",
      nullable: true,
      precision: "22"
    };
    const wrapper = shallow(<StructureField {...structureField} />);
    expect(wrapper).toMatchSnapshot();
  });
});
