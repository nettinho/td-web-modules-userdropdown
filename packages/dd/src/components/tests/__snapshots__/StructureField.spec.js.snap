// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StructureField /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Container
    as={[Function]}
    text={true}
  >
    <Header
      as="h2"
    >
      <HeaderContent>
        f1
      </HeaderContent>
    </Header>
    <Connect(StructureFieldForm) />
    <InjectIntl(Connect(StructureFieldProperties)) />
  </Container>
</React.Fragment>
`;
